File: c/004.c
/**
 * Project Euler: Challenge 004 - http://projecteuler.net/problem=4
 *
 * A palindromic number reads the same both ways.
 * The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
 * Find the largest palindrome made from the product of two 3-digit numbers.
 *
 * Answer: 906609
 */
#include <stdio.h>
#include <stdbool.h>

short int isp(int n){
  int r = 0, t = n;
  while (t != 0) {
   r = (r * 10) + (t%10);
   t /= 10;
  }
  return (n == r);
}

int main(void) {
  int i = 0, j = 0, k = 0, l = 0;
  for (i = 100; i <= 999; i++) {
    for (j = 100; j <= i; j++) {
      l = (i * j);
      k = (isp(l)) && (l > k) ? l : k;
    }
  }
  printf("%i\n", k);
  return 0;
}

File: python/004.py
# Project Euler: Challenge 004 - http://projecteuler.net/problem=4
#
# A palindromic number reads the same both ways. 
# The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
# Find the largest palindrome made from the product of two 3-digit numbers.
# 
# Answer: 906609
#

largest = 0

for n1 in range(100, 1000):
	for n2 in range(101, 1000):
		product = n1  * n2
		str_product = str(product)
		palindromic = str_product == str_product[::-1]
		largest = product if palindromic and product > largest else largest

print(largest)
